stablehlo/reference/Ops.cpp

```diff
@@ -38,6 +38,13 @@ SmallVector<int64_t> permute(ArrayRef<int64_t> array, ArrayRef<int64_t> perm) {
   return result;
 }
 
+SmallVector<int64_t> addIndices(ArrayRef<int64_t> lhs, ArrayRef<int64_t> rhs) {
+  SmallVector<int64_t> combined;
+  for (auto [lhsIdx, rhsIdx] : llvm::zip(lhs, rhs))
+    combined.push_back(lhsIdx + rhsIdx);
+  return combined;
+}
```

Comment:
Given that we've started introducing these helper functions, I think we should create `class Index` and supersede `ArrayRef<int64_t>` for these things in the very near future. Doesn't have to block this PR, but can we open a ticket so that we don't forget? See related discussion in https://github.com/openxla/stablehlo/pull/1005#discussion_r1103409763.

Reply:
Opened #1163.